scaleway/resource_rdb_read_replica.go

```diff
@@ -0,0 +1,300 @@
+package scaleway
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/scaleway/scaleway-sdk-go/api/rdb/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func resourceScalewayRdbReadReplica() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScalewayRdbReadReplicaCreate,
+		ReadContext:   resourceScalewayRdbReadReplicaRead,
+		UpdateContext: resourceScalewayRdbReadReplicaUpdate,
+		DeleteContext: resourceScalewayRdbReadReplicaDelete,
+		Timeouts: &schema.ResourceTimeout{
+			Create:  schema.DefaultTimeout(defaultRdbInstanceTimeout),
+			Read:    schema.DefaultTimeout(defaultRdbInstanceTimeout),
+			Update:  schema.DefaultTimeout(defaultRdbInstanceTimeout),
+			Delete:  schema.DefaultTimeout(defaultRdbInstanceTimeout),
+			Default: schema.DefaultTimeout(defaultRdbInstanceTimeout),
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"instance_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Id of the rdb instance to replicate",
+			},
+			"endpoints": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"direct_access": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							Description: "Direct access endpoint",
+							MaxItems:    1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									// Endpoints common
+									"endpoint_id": {
+										Type:        schema.TypeString,
+										Description: "UUID of the endpoint (UUID format).",
+										Computed:    true,
+									},
+									"ip": {
+										Type:        schema.TypeString,
+										Description: "IPv4 address of the endpoint (IP address). Only one of ip and hostname may be set.",
+										Computed:    true,
+									},
+									"port": {
+										Type:        schema.TypeInt,
+										Description: "TCP port of the endpoint.",
+										Computed:    true,
+									},
+									"name": {
+										Type:        schema.TypeString,
+										Description: "Name of the endpoint.",
+										Computed:    true,
+									},
+									"hostname": {
+										Type:        schema.TypeString,
+										Description: "Hostname of the endpoint. Only one of ip and hostname may be set.",
+										Computed:    true,
+									},
+								},
+							},
+						},
+						"private_network": {
```

Comment:
Can we have `direct_access` and `private_network`? Maybe add Conflict with

Reply:
We can have both